indexing-service/src/main/java/org/apache/druid/indexing/overlord/TaskRunner.java

```diff
@@ -121,4 +121,17 @@ default TaskLocation getTaskLocation(String taskId)
    * @return ScalingStats if the runner has an underlying resource which can scale, Optional.absent() otherwise
    */
   Optional<ScalingStats> getScalingStats();
+
+  /**
+   * APIs useful for emitting statistics for @TaskSlotCountStatsMonitor
+  */
+  long getTotalTaskSlotCount();
```

Comment:
How about adding new interfaces such as `TaskSlotCountable` and `WorkerCountable`? Then, we can avoid implementing unnecessary interfaces, e.g., the new methods in `SingleThreadedBackgroundRunner`, by implementing required interfaces only.